# Assignment 1
### Understanding Uncertainty
### Due 9/5

1. Create a new public repo on Github under your account. Include a readme file.
2. Clone it to your machine. Put this file into that repo.
3. Use the following function to download the example data for the course:

In [1]:
def download_data(force=False):
    """Download and extract course data from Zenodo."""
    import urllib.request, zipfile, os
    
    zip_path = 'data.zip'
    data_dir = 'data'
    
    if not os.path.exists(zip_path) or force:
        print("Downloading course data")
        urllib.request.urlretrieve(
            'https://zenodo.org/records/16954427/files/data.zip?download=1',
            zip_path
        )
        print("Download complete")
    else:
        print("Download file already exists")
        
    if not os.path.exists(data_dir) or force:
        print("Extracting data files...")
        with zipfile.ZipFile(zip_path, 'r') as zip_ref:
            zip_ref.extractall(data_dir)
        print("Data extracted")
    else:
        print("Data directory already exists")

download_data()

Download complete
Extracting data files...
Data extracted


4. Open one of the datasets using Pandas:
    1. `ames_prices.csv`: Housing characteristics and prices
    2. `college_completion.csv`: Public, nonprofit, and for-profit educational institutions, graduation rates, and financial aid
    3. `ForeignGifts_edu.csv`: Monetary and in-kind transfers from foreign entities to U.S. educational institutions
    4. `iowa.csv`: Liquor sales in Iowa, at the transaction level
    5. `metabric.csv`: Cancer patient and outcome data
    6. `mn_police_use_of_force.csv`: Records of physical altercations between Minnessota police and private citizens
    7. `nhanes_data_17_18.csv`: National Health and Nutrition Examination Survey
    8. `tuna.csv`: Yellowfin Tuna Genome (I don't recommend this one; it's just a sequence of G, C, A, T )
    9. `va_procurement.csv`: Public spending by the state of Virginia

In [11]:
import pandas as pd
import numpy as np
import os

In [17]:
os.chdir("/Users/josephhudson/Desktop/DS_5030_HW_Submissions/data")
df=pd.read_csv("iowa.csv")
df.head()

Invoice/Item Number        Date  Store Number                   Store Name  \
0     INV-59108400026  06/06/2023          3723           J D SPIRITS LIQUOR   
1        S16879800057  01/16/2014          3926  LIQUOR DOWNTOWN / IOWA CITY   
2     INV-05301100019  06/05/2017          3829     GARY'S FOODS / MT VERNON   
3     INV-40973500083  10/14/2021          5102               WILKIE LIQUORS   
4     INV-17022500013  01/18/2019          2560   HY-VEE FOOD STORE / MARION   

  Zip Code          Category Name           Vendor Name  Item Number  \
0    51040  STRAIGHT RYE WHISKIES      INFINIUM SPIRITS        27102   
1    52240         VODKA 80 PROOF    HEAVEN HILL BRANDS        35416   
2    52314      CANADIAN WHISKIES       DIAGEO AMERICAS        11296   
3    52314      AMERICAN SCHNAPPS       JIM BEAM BRANDS        82787   
4    52302        WHISKEY LIQUEUR  SAZERAC COMPANY  INC        64863   

            Item Description  Bottle Volume (ml)  State Bottle Cost  \
0          TEMPLETON RYE 4YR                 750              18.09   
1     BURNETT'S VODKA 80 PRF                 750               4.84   
2                CROWN ROYAL                 750              15.59   
3       DEKUYPER BUTTERSHOTS                1000               7.87   
4  FIREBALL CINNAMON WHISKEY                 200               2.50   

   State Bottle Retail  Bottles Sold  Sale (Dollars)  
0                27.14             2           54.28  
1                 7.26            12           87.12  
2                23.39             6          135.66  
3                11.81            12          141.72  
4                 3.75            12           45.00

In [18]:
df.shape

(159904, 14)

5. Pick two or three variables and briefly analyze them
    - Is it a categorical or numeric variable?
    - How many missing values are there? (`df['var'].isna()` and `np.sum()`)
    - If categorical, tabulate the values (`df['var'].value_counts()`) and if numeric, get a summary (`df['var'].describe()`)

## The Vendor Name Variable

The Vendor Name variable is a categorical variable that tells us what company is the distibutior of the alcohol being purchased.


In [13]:
int(np.sum(df["Vendor Name"].isna()))

0

There are 0 missing values in the Vendor Name column.

In [14]:
df["Vendor Name"].value_counts()

Vendor Name
DIAGEO AMERICAS                       26068
JIM BEAM BRANDS                       14163
SAZERAC COMPANY  INC                  13481
HEAVEN HILL BRANDS                     8987
LUXCO INC                              7996
                                      ...  
SUTTER HOME WINERY INC                    1
LLOYD CARUSO LLC                          1
PENNSYLVANIA PURE DISTILLERIES LLC        1
PACIFIC EDGE WINE & SPIRITS               1
THREE RANGERS                             1
Name: count, Length: 341, dtype: int64

In [19]:
(26068+14163+13481+8987+7996)/159904

0.4421090154092455

It appears that of the 155904 sales in this dataset, around 44% of the sales involve the venders of Diageo Americas, Jim Beam Brands, Sazerac Comapny Inc, Heaven Hill Brands, and Luxico Inc.

## The Bottle Volume Variable

The Bottle Volume Variable is a numerical variable that tells us how many milliliters each bottle has that is in the sale.

In [21]:
int(np.sum(df["Bottle Volume (ml)"].isna()))

0

There are 0 missing values in the Bottle Volume column.

In [22]:
df["Bottle Volume (ml)"].describe()

count    159904.000000
mean        869.592737
std         513.812818
min          20.000000
25%         750.000000
50%         750.000000
75%        1000.000000
max        6000.000000
Name: Bottle Volume (ml), dtype: float64

Based on the quantiles displayed, as well as mean of 869.593, we can determine that a large amount of the sales included bottles of 750 ml volumes. 

6. What are some questions and prediction tools you could create using these data? Who would the stakeholder be for that prediction tool? What practical or ethical questions would it create? What other data would you want, that are not available in your data?

One question would be what connection is there between the profit margin of a particular bottle of alcohol and how many of these bottles are sold. With this data, we can see both State bottle cost and the actual retail sale price, allowing us to calculate how much profit is made off each sold bottle of a particular item. In theory, bottles with higher differences in state cost and retail price should be sold less, given how much their price is marked up. We could create a scatter plot to view the relationship between the 2 numeric variables, and (if the assumptions are met) we could use a simple linear regression model to see how much of the variation in the total bottles sold of a item is explained by the item's difference in State cost and retail price. This information could be useful for the State officials who run the ABC stores in their state when it comes to pricing items in hopes of increasing profit by potentially having more bottles sold at lower prices. This would create a larger economic question of how much the demand for a particular item would actually change based on the retail price change of the item, and if it would actually help increase profits for the state. To better answer this question, I would also like data on how similar items are priced in other ABC states and to compare how Iowa prices the same products as another ABC state and the difference in total bottles sold. This comparison would obviously need to in some way take into account the different populations of states. 

Another question could be which places in Iowa have the most frequent liquor sales/highest sum of sales, and based off this data you could make a table showing the zip codes where the highest sum of sales/most sales happen. A stakeholder who is trying to plan where they want to build their next liquor store might use that information to then choose a location. The ethical concern with this approach comes back to the somewhat predatory nature that is present with a liquor store using the information onf where most lqiuor is bought to then put more stores there and thus allow people who buy liquor frequently even more options of consuming it. Some data that is not included in this dataset that might be useful would be the towns the liquor sales are taking place in. I think it would then be interesting to have the estimated poulations of these towns and zip code to then see sales per capita for each area to better identify which places have people on average that are more likely to buy liquor based on past purchases. 

7. Commit your work to the repo (`git commit -am 'Finish assignment'` at the command line, or use the Git panel in VS Code). Push your work back to Github and submit the link on Canvas in the assignment tab.